recipes/vc/all/conanfile.py

```diff
@@ -18,14 +19,31 @@ class VcConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "fPIC": [True, False],
+        # https://github.com/VcDevel/Vc/blob/1.4.4/cmake/OptimizeForArchitecture.cmake
+        "cpu_architecture": [
+            "auto", "generic", "none", "x86-64", "x86-64-v2", "x86-64-v3", "x86-64-v4",
+            # Intel
+            "core", "merom", "penryn", "nehalem", "westmere", "sandy-bridge", "ivy-bridge", "haswell",
+            "broadwell", "skylake", "skylake-xeon", "kaby-lake", "cannonlake", "silvermont", "goldmont",
+            "knl", "atom",
+            # AMD
+            "k8", "k8-sse3", "barcelona", "istanbul", "magny-cours", "bulldozer", "interlagos",
+            "piledriver", "AMD 14h", "AMD 16h", "zen", "zen3"
+        ],
```

Comment:
Could this not be detected from the host profile? Wouldn't that make a bit more sense? It might not be possible but just thinking out loud

Reply:
Hmm, that's a very good point. Not with the current settings, but this could well be a separate `arch.microarchitecture` or something similar. It could even set the appropriate `-march=...` automatically, probably.
Many projects provide an option to set `-march=native`, but this always gets disabled by package managers to build for the lowest common denominator. This is a place where the precise host profile capabilities of Conan could really shine.
Also, duplicating these options across packages is not ideal either. I applied the exact same logic in the OpenMVG recipe as well: #23836.